framework/web/Request.php

```diff
@@ -874,7 +944,7 @@ public function getQueryString()
     public function getIsSecureConnection()
     {
         return isset($_SERVER['HTTPS']) && (strcasecmp($_SERVER['HTTPS'], 'on') === 0 || $_SERVER['HTTPS'] == 1)
-            || isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0;
+            || strcasecmp($this->headers->get('X-Forwarded-Proto'), 'https') === 0;
```

Comment:
If there's no header, `null` will be returned and HHVM would produce "Warning: strcasecmp() expects parameter 1 to be string, null given" error. Either there should be `->has()` check or a default value as an empty string.